api/src/lib/interface.ts

```diff
@@ -670,15 +670,16 @@ export interface OrcidAffiliationDate {
  * - Qualifications
  */
 export interface MappedOrcidAffiliation {
-    id: number;
+    id: number | string;
```

Comment:
I suggest choosing one or the other, but since affiliations coming from ORCID already have a numeric id, and that's what we're going to use from now on, maybe we can generate numeric ids for the old affiliations as well. Keep in mind that this interface is also used on the frontend.